content/en/developers/ide_plugins/idea/code_security.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Code Security
+type: documentation
+further_reading:
+- link: "/security/code_security/"
+  tag: "Documentation"
+  text: "Learn more about Code Security"
+---
+
+## Overview
+The [Code Security][4] integration in the plugin promotes better security by:
+
+- flagging library vulnerabilities and runtime code vulnerabilities
+- checking your code changes locally, as you edit, to detect quality and security issues prior to commit
+
+The feature is supported for any source code repository that is onboarded for [Datadog Code Security][1].
+
+## Vulnerabilities
+The Datadog plugin reports library and runtime code vulnerabilities highlighting the issues directly in your source editor:
+
+{{< img src="/developers/ide_plugins/idea/code_security/library-vulnerability.png" alt="A library vulnerability highlighted in the source editor" style="width:80%;" >}}
+
+Full details of each vulnerability are shown in the Datadog tool-window in the **File Insights** and **Project Insights** tabs. 
+
+{{< img src="/developers/ide_plugins/idea/code_security/library-vulnerability-tool-window.png" alt="A library vulnerability shown in the Datadog tool-window" style="width:100%;" >}}
+
+Click on the link in the `Code Links` section to navigate to the source location, or click on the vulnerability description to open the summary at Datadog.
+
+## Local code analysis
+
+### File editing
+As you edit your source files, the Datadog plugin checks the content (locally) against a set of [rules][2] to detect and flag quality and security issues before you commit your changes.
+
+The local analysis engine supports all the file types listed in [Static Analysis Rules][3]. Issues are shown in the source code editor with the JetBrains inspection system, and suggested fixes can be applied directly:
+
+{{< img src="/developers/ide_plugins/idea/code_security/local-code-analysis.png" alt="Static analysis violation in the source editor" style="width:100%;" >}}
+
+Additionally, all issues detected by this feature are listed in the standard **Problems** view.
+
+### Analyzing multiple files
+You can run the local code analysis across multiple files. In the **Code** menu, select **Analyse Code** → **Run Inspection By Name…** and choose `Datadog Static Analysis`:
+
+{{< img src="/developers/ide_plugins/idea/code_security/inspection-by-name.png" alt="Datadog Static Analysis inspection" style="width:60%;" >}}
+
+Choose the scope, click **OK**, then review the flagged issues in the **Problems** view.
+
+### Configuration
+The code analyzer will run automatically if:
+
+1. your repository has been onboarded for [Datadog Code Security][1]
+2. you are logged in to Datadog so that the remote configuration data is available
```

Comment:
```suggestion
The code analyzer runs automatically if you meet both of these conditions:

- Your repository has been onboarded for [Datadog Code Security][1].
- You are logged in to Datadog so that the remote configuration data is available.
```